functionalDB/chapter.md

```diff
@@ -570,69 +577,67 @@ That transformation occurs in the following transaction call chain:
 
-Transaction: 
+The above process can be seen in the following examples.
 
-User call: 
+For Transaction, the user call: 
 ````clojure
 (transact db-conn  (add-entity e1) (update-entity e2 atr2 val2 :db/add))  
 ````
-Changes into: 
+changes into: 
 ````clojure
 (_transact db-conn swap! (add-entity e1) (update-entity e2 atr2 val2 :db/add))
 ````
-Becomes: 
+which becomes: 
 ````clojure
 (swap! db-conn transact-on-db [[add-entity e1][update-entity e2 atr2 val2 :db/add]])
 ````
-What-if: 
 
-User call: 
+The what-if user call:
+
 ````clojure
 (what-if my-db (add-entity e3) (remove-entity e4))
 ````
-Changes into: 
+changes into: 
 ````clojure
 (_transact my-db _what-if (add-entity e3) (remove-entity e4))
 ````
-Changes into: 
+then:
 ````clojure
 (_what-if my-db transact-on-db [[add-entity e3] [remove-entity e4]])
 ````
-Becomes eventually: 
+and eventually: 
 ````clojure
 (transact-on-db my-db  [[add-entity e3] [remove-entity e4]])
 ````
 
-## Insight extraction as libraries
+## Insight Extraction as Libraries
 
-At this point, we have the core functionality of the database in place, and it is time to add to it its raison d'être - insights extraction. The architecture approach we used here is to allow adding these capabilities as libraries, as different usages of the database would need different such mechanisms. 
+At this point we have the core functionality of the database in place, and it is time to add its *raison d'être*: insights extraction. The architecture approach we used here is to allow adding these capabilities as libraries, as different usages of the database would need different such mechanisms. 
 
-### Graph traversal
+### Graph Traversal
 
-A reference connection between entities is created when an entity’s attribute’s type is *:db/ref*, which means that the value of that attribute is an id of another entity. When a referring entity is added to the database, the reference is indexed at the VAET index.  
-The information found in the VAET index can be leveraged to extract all the incoming links to an entity and is done in the function *incoming-refs*, which collects for the given entity all the leaves that are reachable from it at that index:
+A reference connection between entities is created when an entity’s attribute’s type is `:db/ref`, which means that the value of that attribute is an ID of another entity. When a referring entity is added to the database, the reference is indexed at the VAET index.  
+The information found in the VAET index can be leveraged to extract all the incoming links to an entity. This is done in the `incoming-refs` function, which collects all the leaves that are reachable from the entity at that index:

```

Comment:
I think that between this general description and the code itself, the meaning is made clear enough.